ibm/resource_ibm_private_dns_glb_monitor.go

```diff
@@ -0,0 +1,460 @@
+		ValidateSchema{
+			Identifier:                 pdnsGlbMonitorType,
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              monitorCheckTypes})
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 pdnsGlbMonitorMethod,
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              methods})
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 pdnsGlbMonitorExpectedCodes,
+			ValidateFunctionIdentifier: ValidateAllowedStringValue,
+			Type:                       TypeString,
+			Required:                   true,
+			AllowedValues:              expectedcode})
+	dnsMonitorValidator := ResourceValidator{ResourceName: ibmDNSGlbMonitor, Schema: validateSchema}
+	return &dnsMonitorValidator
+}
+
+func resourceIBMPrivateDNSGLBMonitorCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+	instanceID := d.Get(pdnsInstanceID).(string)
+	CreateMonitorOptions := sess.NewCreateMonitorOptions(instanceID)
+
+	monitorname := d.Get(pdnsGlbMonitorName).(string)
+	monitorinterval := int64(d.Get(pdnsGlbMonitorInterval).(int))
+	monitorretries := int64(d.Get(pdnsGlbMonitorRetries).(int))
+	monitortimeout := int64(d.Get(pdnsGlbMonitorTimeout).(int))
+	CreateMonitorOptions.SetName(monitorname)
+	CreateMonitorOptions.SetInterval(monitorinterval)
+	CreateMonitorOptions.SetRetries(monitorretries)
+	CreateMonitorOptions.SetTimeout(monitortimeout)
+	if monitordescription, ok := d.GetOk(pdnsGlbMonitorDescription); ok {
+		CreateMonitorOptions.SetDescription(monitordescription.(string))
+	}
+	if Mtype, ok := d.GetOk(pdnsGlbMonitorType); ok {
+		CreateMonitorOptions.SetType(Mtype.(string))
+	}
+	if Mport, ok := d.GetOk(pdnsGlbMonitorPort); ok {
+		CreateMonitorOptions.SetPort(int64(Mport.(int)))
+	}
+	if monitorpath, ok := d.GetOk(pdnsGlbMonitorPath); ok {
+		CreateMonitorOptions.SetPath((monitorpath).(string))
+	}
+	if monitorexpectedcodes, ok := d.GetOk(pdnsGlbMonitorExpectedCodes); ok {
+		CreateMonitorOptions.SetExpectedCodes((monitorexpectedcodes).(string))
+	}
+	if monitormethod, ok := d.GetOk(pdnsGlbMonitorMethod); ok {
+		CreateMonitorOptions.SetMethod((monitormethod).(string))
+	}
+	if monitorexpectedbody, ok := d.GetOk(pdnsGlbMonitorExpectedBody); ok {
+		CreateMonitorOptions.SetExpectedBody((monitorexpectedbody).(string))
+	}
+	if monitorheaders, ok := d.GetOk(pdnsGlbMonitorHeaders); ok {
+		expandedmonitorheaders, err := expandGLBMonitors(monitorheaders)
+		if err != nil {
+			return err
+		}
+		CreateMonitorOptions.SetHeadersVar(expandedmonitorheaders)
+	}
+	if monitorallowinsecure, ok := d.GetOk(pdnsGlbMonitorAllowInsecure); ok {
```

Comment:
instead of d.GetOk(), use d.GetOkExists() - this is applicable for only bool attribute

Reply:
done